drivers/dma/dw-axi-dmac/dw-axi-dmac-platform.c

```diff
@@ -918,6 +918,9 @@ dw_axi_dma_chan_prep_slave_sg(struct dma_chan *dchan, struct scatterlist *sgl,
 		num_segments = DIV_ROUND_UP(sg_dma_len(sg), axi_block_len);
 		segment_len = DIV_ROUND_UP(sg_dma_len(sg), num_segments);
```

Comment:
If num_segments can be on the next line, why didn't we get a division-by-zero error here?

Reply:
Looking at a disassembly of the output, the compiler seems to have moved the `(num_segments == 0)` test (which should by `(!num_segments)` BTW) between the two DIV_ROUND_UP calls, hence the lack of an exception.

Reply:
There was no divide by zero exception before this PR where the `num_segments == 0` protection wasn't present, nor with logging added which printed the result of the divide by zero (which was 0).
For interest, @pelwell did find:
```
  ARM Architecture Reference Manual, ARMv8 [ARM DDI 0487B.b]
      http://infocenter.arm.com/help/topic/com.arm.doc.ddi0487b.b/DDI0487B_b_armv8_arm.pdf
      1) aarch64 instruction set: section C3.4.7 and C6.2.279 (UDIV)
         "A division by zero results in a zero being written to
          the destination register, without any indication that
          the division by zero occurred."
      2) aarch32 instruction set: section F1.4.8 and F5.1.263 (UDIV)
         "For the SDIV and UDIV instructions, division by zero
          always returns a zero result."
```

Reply:
The !num_segments test should go before the segment_len calculation, if only for appearances, but otherwise I agree with this commit and the reasoning behind it.